pjrt/convert/utils/xplane_schema.zig

```diff
@@ -0,0 +1,300 @@
+    @"IteratorGetNextAsOptionalOp::DoCompute",
+    Iterator,
+    @"Iterator::Prefetch::Generator",
+    PrefetchProduce,
+    PrefetchConsume,
+    ParallelInterleaveProduce,
+    ParallelInterleaveConsume,
+    ParallelInterleaveInitializeInput,
+    ParallelMapProduce,
+    ParallelMapConsume,
+    MapAndBatchProduce,
+    MapAndBatchConsume,
+    ParseExampleProduce,
+    ParseExampleConsume,
+    ParallelBatchProduce,
+    ParallelBatchConsume,
+    // Batching related.
+    BatchingSessionRun,
+    ProcessBatch,
+    BrainSessionRun,
+    ConcatInputTensors,
+    MergeInputTensors,
+    ScheduleWithoutSplit,
+    ScheduleWithSplit,
+    ScheduleWithEagerSplit,
+    @"ASBSQueue::Schedule",
+    // TFRT related.
+    TfrtModelRun,
+    // Serving related.
+    ServingModelRun,
+    // GPU related.
+    KernelLaunch,
+    KernelExecute,
+    // TPU related
+    EnqueueRequestLocked,
+    RunProgramRequest,
+    HostCallbackRequest,
+    TransferH2DRequest,
+    TransferPreprocessedH2DRequest,
+    TransferD2HRequest,
+    OnDeviceSendRequest,
+    OnDeviceRecvRequest,
+    OnDeviceSendRecvLocalRequest,
+    CustomWait,
+    OnDeviceSendRequestMulti,
+    OnDeviceRecvRequestMulti,
+    PjrtAsyncWait,
+    DoEnqueueProgram,
+    DoEnqueueContinuationProgram,
+    WriteHbm,
+    ReadHbm,
+    TpuExecuteOp,
+    CompleteCallbacks,
+    @"tpu::System::TransferToDevice=>IssueEvent",
+    @"tpu::System::TransferToDevice=>IssueEvent=>Done",
+    @"tpu::System::TransferFromDevice=>IssueEvent",
+    @"tpu::System::TransferFromDevice=>IssueEvent=>Done",
+    @"tpu::System::Execute",
+    @"TPUPartitionedCallOp-InitializeVarOnTPU",
+    @"TPUPartitionedCallOp-ExecuteRemote",
+    @"TPUPartitionedCallOp-ExecuteLocal",
+    Linearize,
+    Delinearize,
+    @"TransferBufferFromDevice-FastPath",
+
+    pub fn fromString(event_name: []const u8) ?HostEventType {
+        return std.meta.stringToEnum(HostEventType, event_name);
+    }
+
+    pub fn fromTfOpEventType(event_name: []const u8) ?HostEventType {
+        return switch (tf_op_utils.parseTfOpCategory(event_name)) {
+            .tensorflow => .TfOpRun,
+            .tf_data => .Iterator,
+            else => null,
+        };
+    }
+
+    pub fn isInternalEvent(event_type: HostEventType) bool {
+        // TODO(b/162102421): Introduce a prefix for internal event names.
+        return switch (event_type) {
+            .MemoryAllocation,
+            .MemoryDeallocation,
+            .PrefetchProduce,
+            .PrefetchConsume,
+            .ParallelInterleaveProduce,
+            .ParallelInterleaveConsume,
+            .ParallelInterleaveInitializeInput,
+            .ParallelMapProduce,
+            .ParallelMapConsume,
+            .MapAndBatchProduce,
+            .MapAndBatchConsume,
+            .ParseExampleProduce,
+            .ParseExampleConsume,
+            => true,
+            else => false,
+        };
+    }
+};
+
+pub const StatType = enum(u16) {
+    UnknownStatType = 0,
```

Comment:
just use "unknown", the type system knows it's a StateType

Reply:
When initializing `XPlaneVisitor`, we need to be able to convert from the string representation of the enum value to the actual zig `StatType` enum type. XLA has the string representation for unknown ops as `"UnknownStatType"`. 

For simplicity of the `StatType.fromString` logic, imo it's best to leave as is so we can purely leverage `std.meta.stringToEnum`.

Reply:
oh, didn't realize this could happen in protobuf too, I thought it was your fallback for unhandled protobuffer fields